README.md

```diff
@@ -123,6 +123,26 @@ let x = 0.0
 end
 ```
 
+Note that the let block is used here to create a scope to allow assignment of 
+`x` within `@byrow!`.
+
+`byrow!` also supports special syntax for allocating new columns to make
+`byrow!` more useful for data transformations. The syntax `@newcol
+x::Array{Int}` allocates a new column `:x` with an `Array` container with eltype
+`Int`. Note that the returned AbstractDataFrame includes these new columns, but
```

Comment:
Yeah, I had thought about doing that. But that would introduce type instability for typed data frames, so it should only be allowed for the current `DataFrame` to avoid trapping users. Maybe better keep it as you wrote it, and possibly revise this later.